src/extension.ts

```diff
@@ -9,6 +9,7 @@
  *********************************************************************/
 import { ExtensionContext, commands, window } from 'vscode';
 import { MemoryServer } from './memory/server/MemoryServer';
+export { MemoryServer } from './memory/server/MemoryServer';
```

Comment:
Why are you exporting this from `extension.ts`?

Reply:
I've exported this in order to be extended. At first, I've only exported this in index.ts, but the class was not recognized as exported until this change was made.

Reply:
Ok found why: https://github.com/eclipse-cdt/cdt-gdb-vscode/blob/0e3779a19addae1c86ee3afb1514e420c6457d8c/package.json#L17

Reply:
@paul-marechal - does this mean you are ok with this part of the change? What about the rest of the change?

Reply:
The changes looks ok, although I'm just curious about how @alicetrifu is actually extending this extension? It would help to see if there is a better approach or not.

Reply:
Thanks @paul-marechal - I look forward to @alicetrifu answering some of these questions.

Reply:
We need to create a vsix file specific to us. It will use the cdt-gdb-adpater extension,  as well, which will also contain, additionally, commands specific to the targets we using. By extending this, we will be able to have an extension that will contain specific information about the debugging process that users will be able to do on our supported targets.

Reply:
Thanks @alicetrifu - this appears to be fully resolved now. @paul-marechal if you have any further feedback we can follow up in a subsequent issue/pr.